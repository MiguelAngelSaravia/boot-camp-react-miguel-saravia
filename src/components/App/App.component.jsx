import React from 'react';
import { BrowserRouter , Switch, Route } from 'react-router-dom';

import HomePage from '../../pages/Home';
import LoginPage from '../../pages/Login';
import NotFoundPage from '../../pages/NotFound';
import Favorites from '../../pages/Favorites';
import VideoDetailsView from '../../pages/VideoDetailsView';

function App() {
    return (
        <BrowserRouter>
            <Switch>

                <Route exact path="/" >
                    <LoginPage />
                </Route>
                <Route exact path="/home" >
                    <HomePage />
                </Route>
                <Route exact path="/home/:id" >
                    <VideoDetailsView />
                </Route>
                <Route exact path="/favorites" >
                    <Favorites />
                </Route>
                <Route exact path="*">
                    <NotFoundPage />
                </Route>

            </Switch>
        </BrowserRouter>
    )
}
export default App;