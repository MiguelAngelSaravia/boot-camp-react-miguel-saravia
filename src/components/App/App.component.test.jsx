import React from 'react';
import {shallow} from '../../enzyme'
import App from './';
import AuthProvider from '../../providers/Auth';

describe('App', () => {
    const wrapper = shallow(
        <AuthProvider>
          <App/>
        </AuthProvider>
      );
    
      it('Renders App', () => {
        expect(wrapper.exists()).toBe(true)
      });
      it("Find Component", () => {
        expect(wrapper.find('App').length).toBe(1);
      });
      it("Snapshot Component", () => {
        expect(wrapper.find('App').length).toMatchSnapshot();
      });
});
