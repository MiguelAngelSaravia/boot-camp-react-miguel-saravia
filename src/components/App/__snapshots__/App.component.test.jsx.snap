// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot Component 1`] = `1`;
